Source/Xamarin/Prism.Forms/Navigation/Xaml/NavigateToExtension.cs

```diff
@@ -0,0 +1,26 @@
+﻿using Xamarin.Forms;
+
+namespace Prism.Navigation.Xaml
+{
+    [ContentProperty(nameof(Name))]
+    public class NavigateToExtension : Navigation
+    {
+        public bool Animated { get; set; } = true;
+        public string Name { get; set; }
+        public bool? UseModalNavigation { get; set; } = null;
+
+        public override async void Execute(object parameter)
+        {
+            var parameters = parameter.ToNavigationParameters(Bindable);
+             
+            IsNavigating = true;
+            RaiseCanExecuteChanged();
+            
+            var navigationService = GetNavigationService(Page);
+            await navigationService.NavigateAsync(Name, parameters, UseModalNavigation, Animated);
```

Comment:
let's refactor this to:

```cs
protected virtual Task HandleNavigation(string uri, NavigationParameters parameters)
```

this will help make the extension more Extensible so that others could inject some custom handlers if the NavigationResult indicates it was unsuccessful.

Reply:
If a user inherits this class, they've already got access to "Name" so the `uri` parameter is probably not necessary, What I did was pass in the parameters and the actual NavigationService. That way they have access to the service within their override.